app/src/main/java/be/digitalia/mediasession2mqtt/MainWorker.kt

```diff
@@ -79,12 +84,70 @@ class MainWorker @Inject constructor(
         }
     }
 
+    private suspend fun publishHassConfiguration(
+        client: MQTTPublishClient,
+        qosLevel: MQTTQoSLevel,
+        deviceId: Int
+    ) {
+        // Using as a unique identifier, but this may not be available on all devices.
+        val serialNumber = Build.SERIAL
```

Comment:
`Build.SERIAL` will always return "UNKNOWN" for devices running Android 8 or later (approx. 96% of devices today) and only preinstalled system apps may retrieve the serial after that.
If possible you should skip providing this value entirely.
If a serial is mandatory for Hass discovery to work properly, you can use `Settings.Secure.getString(contentResolver, Settings.Secure.ANDROID_ID)`. This value is a 64-bit hexadecimal number, unique per device and per user, but changes when switching to another user or after a factory reset of the device.

Reply:
Good to know! I removed `SERIAL` in favor of `deviceId`, there is no real need to show serial in homeassistant. The unique ids are how homeassistant matches devices and entities, such as serial number or a mac address, so my initial thought was to use `SERIAL` even when I saw the warnings.

Reply:
`deviceId` is a simple number like 1, 2, 3. Could it cause a collision problem? If so, the app should use a GUID like ANDROID_ID or something generated locally in the app.